src/server/status_server/mod.rs

```diff
@@ -495,6 +502,36 @@ where
         }
     }
 
+    fn handle_get_metrics(req: Request<Body>) -> hyper::Result<Response<Body>> {
+        // the following logic is port from prometheus's golang:
+        // https://github.com/prometheus/client_golang/blob/24172847e35ba46025c49d90b8846b59eb5d9ead/prometheus/promhttp/http.go#L155-L176
+        let encoding = req
```

Comment:
This may be extracted as a function `client_accept_gzip`.

Reply:
done